subsys/net/lib/lwm2m/lwm2m_shell.c

```diff
@@ -466,6 +470,65 @@ static int cmd_unlock(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_cache(const struct shell *sh, size_t argc, char **argv)
+{
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
+	int rc;
+	int elems;
+	char *path;
+	struct lwm2m_time_series_elem *cache;
+	struct lwm2m_obj_path obj_path;
+
+	if (argc != 3) {
+		shell_error(sh, "wrong parameters\n");
+		return -EINVAL;
+	}
+
+	/* translate path -> path_obj */
+	rc = lwm2m_string_to_path(argv[1], &obj_path, '/');
+	if (rc < 0) {
+		return rc;
+	}
+
+	if (obj_path.level < 3) {
```

Comment:
```suggestion
	if (obj_path.level < LWM2M_PATH_LEVEL_RESOURCE) {
```

Reply:
Actually, this whole check should be removed. `lwm2m_engine_enable_cache()` is (and should) do the path validation.
Shell commands should not try to be clever, these are for testing, so we need to be able to feed in a bad data as well.